### PR TITLE
Create Dockerfile

### PR DESCRIPTION
**Pull request by mteodori**
Add a Dockerfile to publish an image to Docker Hub for easier usage with pre-commit and tools that support launching commands via docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM golang:1.13.15-alpine AS builder
```
Comment (rikatz) [inline Dockerfile:1]: I'll try to migrate to Go 1.15, if you want to already give a try :)

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+RUN apk add --update --no-cache ca-certificates bash make gcc musl-dev git openssh wget curl
```
Comment (rikatz) [inline Dockerfile:3]: I don't think the builder container would need anything other than the compiler :)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+WORKDIR /go/src/kubepug
+
+COPY go.mod .
```
Comment (rikatz) [inline Dockerfile:7]: Can we make the whole copy as one line? This would generate less layers. Like COPY . /go/src/kubepug (as an example).

**Comment by rikatz** [comment]
@mteodori thank you!

I've left some reviews in the Dockerfile.

Can you also provide an example in the README on how to use this image?

Thanks

**Comment by mteodori** [comment]
sorry @rikatz totally forgot this and with the check breaking I don't understand I gave up

**Comment by rikatz** [comment]
No problem.

Will re-open and take a look :)

**Pull request closed without merging**